##Importing libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

##Importing dataset

In [2]:
dataset = pd.read_csv("/content/breast_cancer.csv")
X = dataset.iloc[:,1:-1].values
y = dataset.iloc[:,-1].values

In [3]:
print(dataset.shape)

(683, 11)


In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 683 entries, 0 to 682
Data columns (total 11 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   Sample code number           683 non-null    int64
 1   Clump Thickness              683 non-null    int64
 2   Uniformity of Cell Size      683 non-null    int64
 3   Uniformity of Cell Shape     683 non-null    int64
 4   Marginal Adhesion            683 non-null    int64
 5   Single Epithelial Cell Size  683 non-null    int64
 6   Bare Nuclei                  683 non-null    int64
 7   Bland Chromatin              683 non-null    int64
 8   Normal Nucleoli              683 non-null    int64
 9   Mitoses                      683 non-null    int64
 10  Class                        683 non-null    int64
dtypes: int64(11)
memory usage: 58.8 KB


In [5]:
print(dataset.describe())

       Sample code number  Clump Thickness  Uniformity of Cell Size  \
count        6.830000e+02       683.000000               683.000000   
mean         1.076720e+06         4.442167                 3.150805   
std          6.206440e+05         2.820761                 3.065145   
min          6.337500e+04         1.000000                 1.000000   
25%          8.776170e+05         2.000000                 1.000000   
50%          1.171795e+06         4.000000                 1.000000   
75%          1.238705e+06         6.000000                 5.000000   
max          1.345435e+07        10.000000                10.000000   

       Uniformity of Cell Shape  Marginal Adhesion  \
count                683.000000         683.000000   
mean                   3.215227           2.830161   
std                    2.988581           2.864562   
min                    1.000000           1.000000   
25%                    1.000000           1.000000   
50%                    1.000000     

##Split the data for Training and Testing

In [6]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.25,random_state=0)

##Feature Scaling

In [7]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

##Logistic Regression Classification

Training LogisticRegression Model

In [8]:
from sklearn.linear_model import LogisticRegression
classifier1 = LogisticRegression(random_state=0)
classifier1.fit(X_train,y_train)

LogisticRegression(random_state=0)

Creating Confusion Matrix and Accuracy Score

In [9]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier1.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[103   4]
 [  5  59]]
Accuracy: 94.74%


##k_nearest_neighbors

Training k_nearest_neighbors Model

In [10]:
from sklearn.neighbors import KNeighborsClassifier
classifier2 = KNeighborsClassifier(n_neighbors = 5, metric = 'minkowski', p = 2)
classifier2.fit(X_train, y_train)

KNeighborsClassifier()

Creating Confusion Matrix and Accuracy Score

In [11]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier2.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[103   4]
 [  5  59]]
Accuracy: 94.74%


##Support_Vector_Machine

Training Support Vector Machine Model

In [12]:
from sklearn.svm import SVC
classifier3 = SVC(kernel = 'linear', random_state = 0)
classifier3.fit(X_train, y_train)

SVC(kernel='linear', random_state=0)

Creating Confusion Matrix and Accuracy Score

In [13]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier3.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[102   5]
 [  5  59]]
Accuracy: 94.15%


##Kernel_Svm

Training Kernel_svm Model

In [14]:
from sklearn.svm import SVC
classifier4 = SVC(kernel = 'rbf', random_state = 0)
classifier4.fit(X_train, y_train)

SVC(random_state=0)

Creating Confusion Matrix and Accuracy Score

In [15]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier4.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[101   6]
 [  3  61]]
Accuracy: 94.74%


##Naive_Bayes

Training Naive Bayes Model

In [16]:
from sklearn.naive_bayes import GaussianNB
classifier5 = GaussianNB()
classifier5.fit(X_train, y_train)

GaussianNB()

Creating Confusion Matrix and Accuracy Score

In [17]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier5.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[99  8]
 [ 2 62]]
Accuracy: 94.15%


##Random_Forest_Classification

Training Random Forest Classification Model

In [18]:
from sklearn.ensemble import RandomForestClassifier
classifier6 = RandomForestClassifier(n_estimators = 10, criterion = 'entropy', random_state = 0)
classifier6.fit(X_train, y_train)

RandomForestClassifier(criterion='entropy', n_estimators=10, random_state=0)

Creating Confusion Matrix and Accuracy Score

In [19]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier6.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[104   3]
 [  5  59]]
Accuracy: 95.32%


##Decision_Tree_Classification

Training Decision Tree Classification Model

In [20]:
from sklearn.tree import DecisionTreeClassifier
classifier7 = DecisionTreeClassifier(criterion = 'entropy', random_state = 0)
classifier7.fit(X_train, y_train)

DecisionTreeClassifier(criterion='entropy', random_state=0)

Creating Confusion Matrix and Accuracy Score

In [21]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = classifier7.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[104   3]
 [  4  60]]
Accuracy: 95.91%


We got highest accuracy of 95.90% with decision tree classification model.

##XGBoost

In [22]:
from xgboost import XGBClassifier
xg_boost = XGBClassifier()
xg_boost.fit(X_train,y_train)

XGBClassifier()

In [23]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = xg_boost.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[103   4]
 [  5  59]]
Accuracy: 94.74%


##Catboost

In [24]:
!pip install catboost

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 76.6/76.6 MB 9.1 MB/s eta 0:00:00


In [25]:
from catboost import CatBoostClassifier
cat_boost = CatBoostClassifier()
cat_boost.fit(X_train, y_train)

Learning rate set to 0.007741
0:	learn: 0.6767644	total: 49.4ms	remaining: 49.3s
1:	learn: 0.6628456	total: 50.2ms	remaining: 25s
2:	learn: 0.6473687	total: 51.8ms	remaining: 17.2s
3:	learn: 0.6308320	total: 53.6ms	remaining: 13.3s
4:	learn: 0.6181673	total: 55.2ms	remaining: 11s
5:	learn: 0.6023751	total: 56.6ms	remaining: 9.38s
6:	learn: 0.5898699	total: 58.2ms	remaining: 8.26s
7:	learn: 0.5754043	total: 59.4ms	remaining: 7.36s
8:	learn: 0.5623533	total: 60.7ms	remaining: 6.68s
9:	learn: 0.5490909	total: 62.6ms	remaining: 6.2s
10:	learn: 0.5371637	total: 66.2ms	remaining: 5.95s
11:	learn: 0.5274926	total: 68.2ms	remaining: 5.61s
12:	learn: 0.5141159	total: 70ms	remaining: 5.32s
13:	learn: 0.5029282	total: 72.5ms	remaining: 5.1s
14:	learn: 0.4936569	total: 73.8ms	remaining: 4.85s
15:	learn: 0.4833632	total: 75.4ms	remaining: 4.64s
16:	learn: 0.4731238	total: 76.2ms	remaining: 4.4s
17:	learn: 0.4638231	total: 79.6ms	remaining: 4.34s
18:	learn: 0.4534757	total: 81.4ms	remaining: 4.2s
19

In [26]:
from sklearn.metrics import confusion_matrix, accuracy_score
y_pred = cat_boost.predict(X_test)
cm = confusion_matrix(y_test, y_pred)
print(cm)
print('Accuracy: {:.2f}%'.format(accuracy_score(y_test, y_pred)*100))

[[103   4]
 [  5  59]]
Accuracy: 94.74%


##K-fold cross validation


In [28]:
from sklearn.model_selection import cross_val_score
accuracies = cross_val_score(estimator = classifier4, X = X_train, y = y_train, cv = 10) #Kernel_Svm
print("Accuracy: {:.2f} %".format(accuracies.mean()*100))
print("Standard Deviation: {:.2f} %".format(accuracies.std()*100))

Accuracy: 97.47 %
Standard Deviation: 2.14 %


We get best accuracy of 97.47% using Kernel_Svm. 